src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java

```diff
@@ -2272,7 +2272,8 @@ public List<Setting<?>> getSettings() {
                     ConfigConstants.USER_ATTRIBUTE_SERIALIZATION_ENABLED,
                     ConfigConstants.USER_ATTRIBUTE_SERIALIZATION_ENABLED_DEFAULT,
                     Property.NodeScope,
-                    Property.Filtered
+                    Property.Filtered,
+                    Property.Dynamic
```

Comment:
Thanks, @markdboyd, for the dynamic implementation.

Quick question: Could toggling this property at runtime cause issues with in-flight requests or active scroll queries? The user context format change might lead to deserialization failures or inconsistent behavior for operations that span the configuration change.

Reply:
@kumjiten I don't think scroll would be affected. The user attributes are only needed for a subset of actions where the plugin persists these values in their system index. I suppose one enhancement could be to start collecting a list of those action names, but one problem with that is that it more tightly couples security with other plugins. Another enhancement could be to configure a list of attributes names that should be serialized and only serialize those attributes without serializing all attributes.

I think for anyone wanting to toggle this setting we should allow it. They can toggle back if they don't use DLS with user attributes substitution.

Reply:
Ok, the consumer of https://github.com/opensearch-project/security/blob/eb7153d772e9e00d49d9cb5ffafb33b5f02399fc/src/main/java/org/opensearch/security/privileges/PrivilegesEvaluator.java#L309-L312 `OPENDISTRO_SECURITY_USER_INFO_THREAD_CONTEXT`needs an update to deserialize the attributes?